sdk/core/azure-core/src/main/java/com/azure/core/http/HttpResponse.java

```diff
@@ -145,6 +147,17 @@ public void writeBodyTo(OutputStream outputStream) throws IOException {
         }
     }
 
+    /**
+     * Writes body content to {@link AsynchronousFileChannel}.
+     * @param asynchronousFileChannel {@link AsynchronousFileChannel}.
+     * @param position The position in the file to begin writing the {@code content}.
+     * @return A {@link Mono} which emits a completion status once the body content has been written to the {@link
+     * AsynchronousFileChannel}.
+     */
+    public Mono<Void> writeBodyTo(AsynchronousFileChannel asynchronousFileChannel, long position) {
```

Comment:
Pending the discussion on `AsynchronousFileChannel` vs `FileChannel`, I wonder if there is a way to make this more inclusive with an abstraction that would allow for many different destination formats to be supplied, `OutputStream`, network socket, `AsynchronousFileChannel`, `FileChannel`, etc. Or, renaming of this method from `writeBodyTo` to `writeToFile` given this is tied to file writing right now.

Reply:
I'm thinking about:
```diff
      public abstract class HttpResponse implements Closeable {
+         public void writeBodyTo(OutputStream outputStream) throws IOException
+         public void writeBodyTo(WritableByteChannel channel) throws IOException
+         public void writeBodyTo(FileChannel channel, long position) throws IOException
+         public Mono<Void> writeBodyTo(AsynchronousFileChannel asynchronousFileChannel, long position)
      }
```

The reason I'm not pursuing general purpose `writeBodyTo(Consumer<ByteBuffer> consumer)` yet is that those abstractions above have well defined contract, i.e. they state that buffer is consumed when method returns (or when future is completed). `Consumer<ByteBuffer>` might be too tempting to just retain references to buffers for the user - but this then might be mitigated by javadocs. Or maybe we need something like `response.getBodyReader().readTo(file)` to group options in extra type without polluting response surface.